internal/builder/builder.go

```diff
@@ -140,6 +148,11 @@ func constructBuilder(img imgutil.Image, newName string, errOnMissingLabel bool,
 		return nil, err
 	}
 
+	if opts.runImage != "" {
+		metadata.RunImages = []RunImageMetadata{{Image: opts.runImage}}
+		metadata.Stack.RunImage.Image = opts.runImage
```

Comment:
Looking at this again, we need to be consistent in how we handle mirrors information. Right now it is being zeroed out for `metadata.RunImages` but preserved for `metadata.Stack.RunImage` (as `metadata.Stack.RunImage.Mirrors` could still have data).

I propose we look in the existing known run images (`RunImageMetadata.Mirrors`) and if the provided `opts.runImage` is found there then preserve the run image information, otherwise zero it out.

Reply:
Yeah, I did that locally, trying to fix the issues with the acceptance but I didn't push it